docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,185 @@
+# Storage node installation and update on Linux
+
+## Abstract
+
+The design document outlines the steps needed to ease installation and provide auto-update of storage nodes on Linux without the use of Docker.
+
+
+## Background
+
+
+The idea is to have two main components, like for Windows: the storage node binary and a storage node updater that will take care of updating the storage node binary according to the rollout version mechanism.
+
+The parts we need:
+- storagenode as a service
+- a system for updating the storagenode binary aka. the updater (with rollout versioning support)
+- a system for updating the updater 
+- a way to collect the configuration data from the user during the installation
+- packaging to ship the above
+
+Linux systems are not unified as windows is. Hence, we have to make choices that will allow us to cover most linux users, with special regards to the raspbian system.
+
+Most commonly used linux distributions seem to be (there are no reliable stats):
+- debian-based (Debian, Ubuntu, Mint, Kali, Raspbian)
+- Red Hat-based (Fedora)
+- Arch Linux (Manjaro is on the rise according to [distrowatch](https://distrowatch.com/dwres.php?resource=popularity)).
+
+All these distributions are shipped with systemD as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised storage
+  - Identity directory
+  - Storage directory
+- Generate `config.yaml` file with the user configuration.
+
+We choose to reuse the storagenode-updater and the recovery mechanism used in windows. They will be daemonized using systemD. The storagenode updater will auto-update. A recovery will be triggered if the updated updater service fails to restart.
+We will use debconf to retrieve user data. 
+
+The debian package will NOT contain the storagenode and storagenode-updater binaries. They will be downloaded as part of the post-installation script. A separate git repository will be created for holding the debian package.
+
+Once we get a fully working debian package, we can convert it to the RPM format using the fpm tool. There are no debconf-like for RPMs, we will need to implement a post-install script to gather user inputs.
```

Comment:
What's the reason not doing `rpm` directly? What's the effort of using fpm vs. writing rpm directly? (I expect some discussion on this).

Reply:
If I understood fpm well, it should be just a call to fpm and we would get the rpm package. I don't think there is expertise in rpm here (well, I don't have any). But maybe we should try to do it, I don't really know.

Reply:
I believe the tool called `alien` gives better-behaved results for package porting than `fpm`. Consider using that.

As for whether to write the rpm directly, I don't think that needs to be defined by this blueprint. Once we have a working deb, we can try out using `fpm` and `alien` and look at the rpms they generate and see if there are any obvious problems with them. Most likely, a good approach will be to use an automated porting tool to make an rpm, then snapshot that in its own git repo and tweak and manage it as a "normal" rpm from that point on. The most common things that need changing in an auto-ported package are directory locations and names/versions of dependencies.